pkg/compare/compare_test.go

```diff
@@ -662,3 +662,230 @@ metadata:
 		})
 	}
 }
+
+func TestYAMLCmpWithYamlInline(t *testing.T) {
+	tests := []struct {
+		desc string
+		a    string
+		b    string
+		want interface{}
+	}{
+		{
+			desc: "ConfigMap data identical",
+			a: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			b: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			want: ``,
+		},
+		{
+			desc: "ConfigMap data order changed",
+			a: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    kind: ValidatingWebhookConfiguration
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    metadata:
+      name: istio-galley
+      value: foo`,
+			b: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      value: foo
+      name: istio-galley`,
+			want: ``,
+		},
+		{
+			desc: "ConfigMap data value change",
+			a: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta1
+    kind: ValidatingWebhookConfiguration
+    metadata:
+      name: istio-galley`,
+			b: `kind: ConfigMap
+data:
+  validatingwebhookconfiguration.yaml: |-
+    apiVersion: admissionregistration.k8s.io/v1beta2
+    kind: ValidatingWebhookConfiguration
```

Comment:
suggest also reordering here to ensure it doesn't leak into output.

Reply:
Sure, reordered.